doc/roots.md

```diff
@@ -129,153 +149,140 @@ x, f(x)
 
 ```
-fzero(f, 1, order=8)
+fzero(f, 0, order=8)
 ```
 
 Basically the high order oscillation can send the proxy tangent line
-off in nearly random directions.
-
-## Polynomials
-
-The `Polynomials` package provides a type for working with polynomial
-functions that allows many typical polynomial operations to be
-defined. In this context, the `roots` function is used to find the
-roots of a polynomial.
-
+off in nearly random directions. The default method can be fooled here
+too.
 
-For example, 
-
-```
-using Polynomials
-x = poly([0.0])			# (x - 0.0)
-roots((x-1)*(x-2)*(x-3))
-```
+----
 
-As a convenience, this package adds a function interface to `roots`:
-
-```
-f(x) = (x-1)*(x-2)*(x^2 + x + 1)
-roots(f)
-```
-
-
-The `fzeros` function will find the real roots of a univariate polynomial:
+Finally, for many functions, all of these methods need a good initial
+guess. For example, the polynomial function $f(x) = x^5 - x - 1$ has
+its one zero near $1.16$. If we start far from it, convergence may
+happen, but it isn't guaranteed:
 
 ```
-fzeros( (x-1)*(x-2)*(x^2 + x + 1))
+f(x) = x^5 - x - 1
+x0 = 0.1
+fzero(f, x0)
 ```
 
-As with roots, this function be called with a function:
+Whereas, 
 
 ```
-f(x) = x*(x-1)*(x^2 + 1)^4
-fzeros(f)
+fzero(f, x0, order=2)
 ```
 
-The algorithm can have numeric issues when the polynomial degree gets
-too large, or the roots are too close together.
-
-
-The `multroot` function will also find the roots. The algorithm does a
-better job when there are multiple roots, as it implements an
-algorithm that first identifies the multiplicity structure of the
-roots, and then tries to improve these values.
-
-
+A graph shows the issue. We have overlayed a 15 steps of Newton's
+method, the other algorithms being somewhat similar:
 
-```
-multroot((x-1)*(x-2)*(x-3))	# roots, multiplicity
+```nocode
+xs = [x0]
+n = 15
+for i in 1:(n-1) push!(xs, xs[end] - f(xs[end])/D(f)(xs[end])) end
+ys = [zeros(Float64,n)';map(f, xs)'][1:2n]
+xs = xs[repeat(collect(1:n), inner=[2], outer=[1])]
+plot(f, -1.25, 1.5, linewidth=3, legend=false)
+plot!(zero, -1.25, 1.5, linewidth=3)
+plot!(xs, ys)
 ```
 
-The `factor` function provides a more pleasant output.
+Though 15 steps are shown, only a few are discernible, as the function's relative maximum
+causes a trap for this algorithm. Starting to the right of the
+relative minimum -- nearer the zero -- would avoid this trap. The default
+method employs a trick to bounce out of such traps, though it doesn't
+always work.
 
+## Finding more than one zero
 
-The `roots` function degrades as there are multiplicities:
+The bracketing methods suggest a simple algorithm to recover multiple
+zeros: partition an interval into many small sub-intervals. For those
+that bracket a root find the root. This is essentially implemented with `fzeros(f,
+a, b)`. The algorithm has problems with zeros that don't cross the
+$x$-axis and zeros which bunch together. Simple usage is often succesful
```

Comment:
"zeros that don't cross the $x$-axis" feels a bit unclear. Maybe "zeros where the function has the same sign to the left and right of the zero". Not sure if that is better, just an idea :)